bliss/main.py

```diff
@@ -14,50 +15,68 @@
 
 
 def generate(gen_cfg: DictConfig):
-    # it's more efficient to launch multiple independent processes than to use workers
-    simulated_dataset = instantiate(gen_cfg.simulator, num_workers=0)
-
     # create cached_data_path if it doesn't exist
     cached_data_path = Path(gen_cfg.cached_data_path)
     if not cached_data_path.exists():
         cached_data_path.mkdir(parents=True)
-    print("Data will be saved to {}".format(cached_data_path))  # noqa: WPS421
+    logger = logging.getLogger(__name__)
+    logger.info(f"Data will be saved to {cached_data_path}")
 
     # log the Hydra config used to generate data to cached_data_path
     with open(cached_data_path / "hparams.yaml", "w", encoding="utf-8") as f:
         OmegaConf.resolve(gen_cfg)
         OmegaConf.save(gen_cfg, f)
 
-    # n_image_files is technically "n_image_files for this process"
-    process_index = gen_cfg.get("process_index", 0)
-    files_start_idx = process_index * gen_cfg.n_image_files
+    ctx = get_context("fork")
+    args = ((gen_cfg, i) for i in range(gen_cfg.n_image_files))
+    with ctx.Pool(processes=gen_cfg.n_processes) as pool:
+        pool.starmap(generate_one_file, args)
+
+
+def generate_one_file(gen_cfg: DictConfig, file_idx: int):
+    prior = instantiate(gen_cfg.prior)
```

Comment:
@kapnadak @gapatron The BLISS decoder can efficiently generate high-resolution images now. If you want to plug in your galaxy cluster prior, it just adjusting the config files so that `cfg.generate.prior` point to your prior rather than the default one.